oak/server/rust/oak_runtime/src/node/wasm/mod.rs

```diff
@@ -567,11 +567,45 @@ impl WasmInterface {
                 valid_channels.push(channel);
             }
         }
-        let valid_statuses = self.runtime.wait_on_channels(&valid_channels)?;
+
```

Comment:
I suspect the current implementation does not line up with the documentation in https://github.com/project-oak/oak/blob/master/docs/abi.md#wait_on_channels any more, could you update that too? I think that may also make it clear to ourselves what API we are aiming for.

Reply:
Done!